metadata-io/src/main/java/com/linkedin/metadata/search/LineageSearchService.java

```diff
@@ -170,11 +174,15 @@ public LineageSearchResult searchAcrossLineage(
 
     EntityLineageResult lineageResult;
     FreshnessStats freshnessStats = new FreshnessStats().setCached(Boolean.FALSE);
-    if (cachedLineageResult == null
-        || finalOpContext.getSearchContext().getSearchFlags().isSkipCache()) {
+    if (cachedLineageResult == null) {
       lineageResult =
-          _graphService.getLineage(opContext, sourceUrn, direction, 0, MAX_RELATIONSHIPS, maxHops);
```

Comment:
This is the core limitation, the intent appears to be avoiding pagination by trying to use a very large response instead. This means practically at limit of 10k or so and no indication to the caller that part of their graph might be missing during the search.